src/cgame/cg_rocket_datasource.cpp

```diff
@@ -34,6 +34,47 @@ Maryland 20850 USA.
 
 #include "cg_local.h"
 
+static void StripNonPrintingCharacters( std::string& str )
+{
+	str.erase(
```

Comment:
What are you trying to do here? std::isprint is another function that should not be used since it doesn't have clearly specified behavior. Is it intended to delete all non-ASCII unicode?

Reply:
I'm looking for a way to filter out non-printable characters, things like ASCII `bell`, Unicode `ZWSP`, etc.

Reply:
We don't have an API for Unicode printable characters. A filter `c >=0 && c < 32` would work against ASCIIbetical sorting abuse at least

Reply:
Well, I can keep all characters for the printed string, as sorting abuse can be taken care in sorting code.

We still need to strip the non-printing characters to detect empty server names only made of not-printable characters (what the current hackish server does if I'm right).

Reply:
So, even if we don't take care about sortable strings, we still want to strip those non-printable characters here.

Reading this page: https://en.cppreference.com/w/cpp/string/byte/isprint
maybe we would want to use `std::setlocale` with `(unvanquished language).UTF-8`?

Reply:
> Reading this page: https://en.cppreference.com/w/cpp/string/byte/isprint
> maybe we would want to use `std::setlocale` with `(unvanquished language).UTF-8`?

No. That API has no ability to handle Unicode and would just say (the constituent bytes of) any Unichode character is unprintable.